packages/kbn-config-schema/src/internals/index.ts

```diff
@@ -314,7 +314,8 @@ export const internals = Joi.extend([
           for (const [entryKey, entryValue] of value) {
             const { value: validatedEntryKey, error: keyError } = Joi.validate(
               entryKey,
-              params.key
+              params.key,
```

Comment:
__note:__ This fixes the issue that's is a blocker for us (without this we can't have required fields within `schema.recordOf` value schema), I'll prepare a separate PR for kbn/config-schema next week or ask Platform team for a help if it turns out to be more complex than I think.

Reply:
Handled in https://github.com/elastic/kibana/pull/60406.